src/EditorFeatures/Core/Implementation/PasteTracking/PasteTrackingPasteCommandHandler.cs

```diff
@@ -44,23 +47,49 @@ public void ExecuteCommand(PasteCommandArgs args, Action nextCommandHandler, Com
         {
             // Capture the pre-paste caret position
             var caretPosition = args.TextView.GetCaretPoint(args.SubjectBuffer);
-
-            // Allow the pasted text to be inserted and formatted.
-            nextCommandHandler();
-
-            if (!args.SubjectBuffer.CanApplyChangeDocumentToWorkspace())
+            if (!caretPosition.HasValue)
             {
                 return;
             }
 
+            // Allow the pasted text to be inserted and formatted.
+            nextCommandHandler();
+
             // Create a tracking span from the pre-paste caret position that will grow as text is inserted.
             var trackingSpan = caretPosition.Value.Snapshot.CreateTrackingSpan(caretPosition.Value.Position, 0, SpanTrackingMode.EdgeInclusive);
 
             // Applying the post-paste snapshot to the tracking span gives us the span of pasted text.
             var snapshotSpan = trackingSpan.GetSpan(args.SubjectBuffer.CurrentSnapshot);
             var textSpan = TextSpan.FromBounds(snapshotSpan.Start, snapshotSpan.End);
 
+            AddUsingsForPaste(args, executionContext, textSpan);
+
             _pasteTrackingService.RegisterPastedTextSpan(args.SubjectBuffer, textSpan);
         }
+
+        private static void AddUsingsForPaste(PasteCommandArgs args, CommandExecutionContext executionContext, TextSpan textSpan)
+        {
+            var sourceTextContainer = args.SubjectBuffer.AsTextContainer();
+            if (!Workspace.TryGetWorkspace(sourceTextContainer, out var workspace))
+            {
+                return;
+            }
+
+            var documentId = workspace.GetDocumentIdInCurrentContext(sourceTextContainer);
+            var document = workspace.CurrentSolution.GetDocument(documentId);
+
+            if (document is null)
+            {
+                return;
+            }
```

Comment:
we have an extension to do all of this given a SourceTextContainer.  Something like GetCurrentDocument or something like that.